.github/workflows/main.yml

```diff
@@ -0,0 +1,198 @@
+name: gh-ci-checks
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number }}-${{ github.event.type }}
+  cancel-in-progress: true
+on:
+  pull_request:
+    paths:
+      - '**.py'
+  push:
+    branches: [main]
+    paths:
+      - '**.py'
+    tags:
+      - 'v*.*.*'
+  workflow_dispatch:
+
+jobs:
+  style:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+          poetry-version: [1.1.14]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          architecture: 'x64'
+
+      - name: Install Poetry ${{ matrix.poetry-version }}
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: ${{ matrix.poetry-version }}
+
+      - name: Install Poetry Dynamic Versioning Plugin
+        run: pip install poetry-dynamic-versioning
+
+      # TODO: once poetry v1.2 is release, we want to only install dev dependencies
+      - name: Install dependencies
+        run: poetry install
+      
```

Comment:
Note @darthtrevino there is no way that I am aware of to only install what's needed for running the style checks. This ends up installing all dependencies right?

I think there is a fix coming up in v1.2 for poetry, but for now, I suppose this is fine since reading more, poetry does have a lot of nice properties.